apps/meteor/ee/app/livechat-enterprise/server/hooks/handleNextAgentPreferredEvents.js

```diff
@@ -25,6 +25,10 @@ const checkDefaultAgentOnNewRoom = (defaultAgent, defaultGuest) => {
 		return defaultAgent;
 	}
 
+	if (!lastChattedAgentPreferred) {
+		return defaultAgent;
+	}
+
```

Comment:
This changes the current logic when "Contact manager preferred routing" is on but "last preferred agent routing" is off. Based on the current logic, it won't transfer the chat to the contactManager.

Maybe you can add a check for both settings here, like

```
if (!lastChattedAgentPreferred && !contactManagerPreferred) {
		return defaultAgent;
}
```

Reply:
:thinking: I think I'll leave this one where it was before, since seems it affects some functionality